tui/installer/screens.py

```diff
@@ -613,13 +613,20 @@ def select_primary_disk(answers):
 
     tui.screen.popHelpLine()
 
-    # entry contains the 'de' part of the tuple passed in
-    answers['primary-disk'] = entry
+    if button == 'back': return LEFT_BACKWARDS
 
-    if 'installation-to-overwrite' in answers:
-        answers['target-is-sr'] = target_is_sr[answers['primary-disk']]
+    # entry contains the 'de' part of the tuple passed in
+    # determine current usage
+    disk = diskutil.probeDisk(entry)
+    if disk.storage[0]:
+        val = snackutil.ButtonChoiceWindowEx(tui.screen,
+            "Confirm Disk Erasure",
+            sr_overwrite_msg.format(diskutil.getHumanDiskLabel(entry, short=True)),
+            ['Yes', 'No'], default=1)
```

Comment:
Maybe writing a small utility function to reuse these lines? They are the same on below chunk.